app/controllers/posts_controller.rb

```diff
@@ -370,11 +370,18 @@ def editor_setup
   end
 
   def import_thread
-    begin
-      importer = PostImporter.new(params[:dreamwidth_url])
-      importer.import(params[:board_id], current_user.id, section_id: params[:section_id], status: params[:status], threaded: params[:threaded])
-    rescue PostImportError => e
-      flash.now[:error] = e.api_error
+    importer = PostImporter.new(params[:dreamwidth_url])
+    importer.import(import_params, user: current_user)
+    if importer.errors.present?
+      if importer.errors.key?(:username)
+        flash.now[:error] = {
+          message: importer.errors[:base].first,
+          array: importer.errors[:username].flatten
+        }
+      else
+        # if we don't have a username array there'll only be one error message
+        flash.now[:error] = importer.errors.full_messages.first
```

Comment:
To be more general / future compatible, maybe we should use error message and array? Message something like "There was an error importing your thread!" and array just `importer.errors.full_messages`

Reply:
Mostly I would have to work out how to _test_ that, but yes